roles/runtime/tasks/setup_aws.yml

```diff
@@ -60,4 +60,82 @@
       register: __dw_builds_async
       until: __dw_builds_async.finished
       retries: 120
-      delay: 30
\ No newline at end of file
+      delay: 30
+
+    - name: Retrieve CDP DW experiences
+      cloudera.cloud.dw_cluster_info:
+        env: "{{ run__env_name }}"
+      register: run__dw_list
+
+    - name: Create CDP DW Database catalogs
+      when: not __dw_dbc_config.use_default_dbc
+      cloudera.cloud.dw_database_catalog:
+        cluster_id : "{{ run__dw_list.clusters[0].id }}"
+        name: "{{ __dw_dbc_config.name }}"
+        load_demo_data: "{{ __dw_dbc_config.load_demo_data }}"
+        state: present
+        wait: yes
+      async: 3600 # 1 hour timeout
+      poll: 0
+      loop: "{{ run__dw_dbc_configs }}"
+      loop_control:
+        loop_var: __dw_dbc_config
+        label: "{{ __dw_dbc_config.name }}"
+      register: __dw_dbc_builds
+
+    - name: Wait for CDP DW Database catalogs setup to complete
+      when: __dw_dbc_build.ansible_job_id is defined
+      ansible.builtin.async_status:
+        jid: "{{ __dw_dbc_build.ansible_job_id }}"
+      register: __dw_dbc_builds_async
+      until: __dw_dbc_builds_async.finished
+      retries: 120
+      delay: 30
+      loop: "{{ __dw_dbc_builds.results }}"
+      loop_control:
+        loop_var: __dw_dbc_build
+        label: "{{ __dw_dbc_build.__dw_dbc_config.name }}"
+
+    - name: Set CDP DW Database catalog name to id map
+      when: __dw_dbc_build_async.database_catalogs is defined
+      ansible.builtin.set_fact:
+        run__dw_dbc_ids: "{{ run__dw_dbc_ids | default({}) | combine({ __dw_dbc_build_async.database_catalogs[0].name : __dw_dbc_build_async.database_catalogs[0].id}) }}"
+      loop: "{{ __dw_dbc_builds_async.results }}"
+      loop_control:
+        loop_var: __dw_dbc_build_async
+
+    - name: Create CDP DW Virtual warehouse
+      cloudera.cloud.dw_virtual_warehouse:
+        cluster_id: "{{ run__dw_list.clusters[0].id }}"
+        dbc_id: "{{ run__dw_dbc_ids[__dw_vw_config.dbc_name] if not __dw_vw_config.use_default_dbc else run__dw_list.clusters[0].dbcs[0].id}}"
+        type: "{{ __dw_vw_config.type }}"
+        name: "{{ __dw_vw_config.name }}"
+        template: "{{ __dw_vw_config.template }}"
+        autoscaling_min_nodes: "{{ __dw_vw_config.autoscaling_min_nodes | int }}"
```

Comment:
I _think_ the `omit()` that would be set in `initialize` task list should carry through here to the module call.